src/awkward/contents/indexedoptionarray.py

```diff
@@ -1575,6 +1575,10 @@ def _completely_flatten(self, nplike, options):
         else:
             return [self.simplify_optiontype()]
 
+    def _drop_none(self):
+        _, _, outindex = self._nextcarry_outindex(self._nplike)
+        return (outindex, self.project())
```

Comment:
Some implementations of `_drop_none` return a `Content`, while others return a 2-tuple of `(Index, Content)`. That's probably part of the problem.